p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,20 +1,85 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { Card } from '../Shared/components/Card'
 import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import { fieldsList, handleSubmitForm, isFieldValid } from './Form.services'
 
 import "./Form.styles.css"
 
+const formData={
+  name:'',
+  lastname:'',
+  birthdate: '',
+  country:'',
+  password:'',
+
+}
 const Form = () => {
+  const [ formState, setFormState ] = useState(formData);
+  const [hasValue, setHasValue]= useState(true);
+  const [fieldsForm,setFieldForm ]=useState(fieldsList);
+  useEffect(() => {
+      if(fieldsList.length  === fieldsForm.filter((field)=>field.hasvalue).length) 
+       {   
+        setHasValue(false);
+      }else{
+        setHasValue(true);
+      }
+
+  }, [formState])
   
+  const onInputChange = ({ target }) => {
+    const { name, value,type } = target;
+      setFormState({
+        ...formState,
+        [name]: value
+      });
+
+ 
+
+  const newList= fieldsForm.map((field)=>{
```

Comment:
There is not problem, i just want to let you know that you can write it like this:

```
setFieldForm(
  fieldsForm.map((field) =>
    field.name === name && isFieldValid(value)
      ? { ...field, hasvalue: true }
      : field
  )
);
```

Reply:
ok, I understand, in the future exercise I will do it.